x-pack/legacy/plugins/lens/server/usage/visualization_counts.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClient, SavedObjectsFindResponse } from 'src/core/server';
+import { Document } from '../../public/persistence';
+import { LensUsage } from './types';
+
+function getTypeName(doc: Document): string {
+  if (doc.visualizationType === 'lnsXY') {
+    return (doc.state.visualization as { preferredSeriesType: string }).preferredSeriesType;
+  } else {
+    return doc.visualizationType as string;
+  }
+}
+
+function groupByType(response: SavedObjectsFindResponse) {
+  const byVisType: Record<string, number> = {};
+
+  response.saved_objects.forEach(({ attributes }) => {
+    const type = getTypeName(attributes as Document);
+
+    if (byVisType[type]) {
+      byVisType[type] += 1;
+    } else {
+      byVisType[type] = 1;
+    }
+  });
+
+  return byVisType;
+}
+
+export async function getVisualizationCounts(
+  savedObjectsClient: SavedObjectsClient
+): Promise<LensUsage> {
+  const [overall, last30, last90] = await Promise.all([
```

Comment:
You're doing three fetches, when you only need one: the first one. You can change your groupByType function to return overall, 30_days, and 90_days. That may be premature optimization, but it seems worth it. To me, telemetry should be very light-weight and have as low an impact on the system as possible, since it is not a primary feature.

I also think we should filter this down to only fetch the subset of fields that we care about, rather than fetching the entire saved object. I think all that we care about is type, visualizationType, and updated_at, right?